resources/schedule.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2021 Nitric Technologies Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/nitrictech/go-sdk/faas"
+)
+
+func rateSplit(rate string) (int, faas.Frequency, error) {
+	rateParts := strings.Split(rate, " ")
+	if len(rateParts) < 2 {
+		return 0, "", fmt.Errorf("not enough parts to rate expression %s", rate)
+	}
+	rateNum := rateParts[0]
+	rateType := rateParts[1]
+
+	num, err := strconv.Atoi(rateNum)
+	if err != nil {
+		return 0, "", fmt.Errorf("invalid rate expression %s; %w", rate, err)
+	}
+
+	switch rateType {
+	case "minutes":
+		// Every nth minute
+		return num, faas.Frequency(rateType), nil
+	case "hours":
+		// The top of every nth hour
+		return num, faas.Frequency(rateType), nil
+	case "days":
+		// Midnight every nth day
+		return num, faas.Frequency(rateType), nil
+	default:
+		return 0, "", fmt.Errorf("invalid rate expression %s; %s must be one of [minutes, hours, days]", rate, rateType)
+	}
+}
```

Comment:
one optional thing, in the node-sdk the rate splitter also accepts singulars as well.
e.g.
```typescript
schedule("Updates").every("1 days", ...)
// is the same as
schedule("Updates").every("day", ...)
```